### PR TITLE
Tank Autoloader: Loader Options / New Tank Gunner Module

### PR DESCRIPTION
**Pull request by JackTheJackhammer**
## About The Pull Request

So NT recently found an ancient T-90M on Earth and discovered the ancient art of Autoloaders. They did not manage to make a good one.

**ABSTRACT**
Many people have issues with finding loaders for the tank. This seeks to provide a less hassle alternative to a loader through the autoloader.

The autoloader's main design is as such: The gunner is given more up-time in gunner's seat, but reloads will take longer and the tank will have to pull back for longer to reload. Currently also considering making the tank explode upon destruction, as the shells in the autoloader are not inside a blow-out-panel.

Current **preliminary** values are as such:
1.5x faster reload time
3 Second Ready Rack Reload Time
Removes MG/Secondary weapon capability
Cuts crew cap to 2 (or 3)
Unable to unload breech manually
Tank actually blows up catastrophically when blown up (like it does damage, maybe even kills or severely maims crew)

The secondary breech becomes the autoloader ring elevator while the breech becomes uninteractable

**TO DO**

- [ ] Make Autoloader Sprite
- [x] Autoloader Magazine Mechanics
- [x] Autoloader Unloading Capability
- [ ] Add secondary ammo counter for the autoloader
- [x] Mald about 1 indexing in Byond Lists
- [ ] Restrict to only LTB tanks

## Why It's Good For The Game

Not having a loader sucks extremely. Not only is it really annoying to hop in and out of a seat, it practically reduces your uptime by half, making the tank a sitting duck to **_LARGER_** foes. This would at least give tankers a option to better their already terrible luck, at a cost.

Basically the combat effectiveness of the Autoloader is still lesser than that of the loader, and you still have to manually load the mg (the true workhorse of the tank), but at least you will be able to fight.

## Changelog
:cl:
add: added the autoloader module to tanks

/:cl:

### CODE REVIEW

**Comment by JackTheJackhammer** [comment]
Welp I can't edit this PR, tis glitched for now (will try again tmrw)
Currently Done:
Magazine Mechanics
Malding

**Comment by Jackal-boop** [comment]
I'm iffy on this idea. I think that the loader should be a requirement.

**Comment by Pariah919** [comment]
Autoloader could easily be the gimmick of an entire tank. As killing the loading part of the loader just relegates them into a spotter.

**Comment by TiviPlus** [comment]
Btw you need to make sure you're taking a tradeoff by not manually loading
Also idk why this is an interior module but works i guess

**Comment by Lauwma** [comment]
> Autoloader could easily be the gimmick of an entire tank. As killing the loading part of the loader just relegates them into a spotter.

You can already have both a loader and a tank commander since ACs get 2 pamphlets

**Comment by JackTheJackhammer** [comment]
> Btw you need to make sure you're taking a tradeoff by not manually loading Also idk why this is an interior module but works I guess Also Also linked to the above might be neat if you made it easy to add this as a module to vehicles like APC who already have interior swaps or vehicles with no interior

I could probably make this simply an ability thing and make the equip of the autoloader swap the breech out with the autoloader breech. Currently its a interior module as it has to swap out the breech for the autoloader with additional functionality

**Comment by JackTheJackhammer** [comment]
On second thought, this may also be like an all-or-nothing module, replacing the need for a loader (with their fumbling reload times) and the MG for a tank gun that is more deadly. It requires more out-wards teamwork as the tank can no longer support the team while additionally being more of an asset if used right. The tank would also explode, without warning, if it dies, becoming a semi-liability if it were to die (turret toss)

Tldr; remove MG and make it more like a SPG: Slightly faster reload (the reload on the tank is already ridiculous by tank standards so not too much) + ease of use at the cost of defensibility

What do y'all think?

**Comment by Blacklist897** [comment]
make the tank turret toss

**Comment by JackTheJackhammer** [comment]
> make the tank turret toss

am
